src/core/state.go

```diff
@@ -444,11 +447,9 @@ func (state *BuildState) AddOriginalTarget(label BuildLabel, addToList bool) {
 		}
 	}
 	if addToList {
-		// The sets of original targets are duplicated between states for all architectures,
-		// we must add it to all of them to ensure everything sees the same set.
-		for _, s := range state.progress.allStates {
-			s.OriginalTargets = append(s.OriginalTargets, label)
-		}
+		state.progress.originalTargetMutex.Lock()
+		state.progress.originalTargets = append(state.progress.originalTargets, label)
+		state.progress.originalTargetMutex.Unlock()
```

Comment:
Do we not care about other architectures anymore?

Reply:
the `progress` sub-structure is shared between all the states so it doesn't need the loop any more.